test/NuGet.Core.Tests/NuGet.Configuration.Test/SettingsFileParsingTests/CertificateItemTests.cs

```diff
@@ -0,0 +1,164 @@
+</configuration>";
+            var nugetConfigPath = "NuGet.Config";
+            using (var mockBaseDirectory = TestDirectory.Create())
+            {
+                ConfigurationFileTestUtility.CreateConfigurationFile(nugetConfigPath, mockBaseDirectory, config);
+
+                // Act and Assert
+                var ex = Record.Exception(() => new SettingsFile(mockBaseDirectory));
+
+                ex.Should().NotBeNull();
+                ex.Should().BeOfType<NuGetConfigurationException>();
+                ex.Message.Should().Be(string.Format("Unable to parse config file because: Missing required attribute 'allowUntrustedRoot' in element 'certificate'. Path: '{0}'.", Path.Combine(mockBaseDirectory, nugetConfigPath)));
+            }
+        }
+
+        [Fact]
+        public void CertificateItem_WithChildren_Throws()
+        {
+            // Arrange
+            var config = @"
+<configuration>
+    <SectionName>
+        <certificate fingerprint=""abcdefg"" hashAlgorithm=""Sha256"" allowUntrustedRoot=""true"">
+            <add key=""key"" value=""val"" />
+        </certificate>
+    </SectionName>
+</configuration>";
+            var nugetConfigPath = "NuGet.Config";
+            using (var mockBaseDirectory = TestDirectory.Create())
+            {
+                ConfigurationFileTestUtility.CreateConfigurationFile(nugetConfigPath, mockBaseDirectory, config);
+
+                // Act and Assert
+                var ex = Record.Exception(() => new SettingsFile(mockBaseDirectory));
+
+                ex.Should().NotBeNull();
+                ex.Should().BeOfType<NuGetConfigurationException>();
+                ex.Message.Should().Be(string.Format("Error parsing NuGet.Config. Element 'certificate' cannot have descendant elements. Path: '{0}'.", Path.Combine(mockBaseDirectory, nugetConfigPath)));
+            }
+        }
+
+        [Fact]
+        public void CertificateItem_ParsedCorrectly()
+        {
+            // Arrange
+            var config = @"
+<configuration>
+    <SectionName>
+        <certificate fingerprint=""abcdefg"" hashAlgorithm=""Sha256"" allowUntrustedRoot=""true"" />
+    </SectionName>
+</configuration>";
+            var nugetConfigPath = "NuGet.Config";
+            using (var mockBaseDirectory = TestDirectory.Create())
+            {
+                ConfigurationFileTestUtility.CreateConfigurationFile(nugetConfigPath, mockBaseDirectory, config);
+
+                // Act and Assert
+                var settingsFile = new SettingsFile(mockBaseDirectory);
+                var section = settingsFile.GetSection("SectionName");
+                section.Should().NotBeNull();
+
+                section.Items.Count.Should().Be(1);
+                var item = section.Items.First() as CertificateItem;
+
+                var expectedItem = new CertificateItem("abcdefg", Common.HashAlgorithmName.SHA256, allowUntrustedRoot: true);
+                item.DeepEquals(expectedItem).Should().BeTrue();
```

Comment:
The test relies on correctness of `DeepEquals(...)` but there are no tests for `DeepEquals(...)`.  Similarly there are no tests for other members like `Equals(...)`, `ElementName`, and `Clone()`.

This feedback applies to all new settings types you're adding, not just `CertificateItem`.